File: src/RESTworld/RESTworld.Client.AspNetCore/DependencyInjection/ClientSettings.cs
using System;

namespace RESTworld.Client.AspNetCore.DependencyInjection
{
    /// <summary>
    /// Contains the settings which the client requests to discover backend APIs.
    /// </summary>
    public class ClientSettings
    {
        /// <summary>
        /// A collection of all backend available APIs which the client may use.
        /// </summary>
        public ApiUrl[] ApiUrls { get; set; } = Array.Empty<ApiUrl>();
    }
}

File: src/Example/ExampleBlog/ExampleBlog.Client.Angular/ClientApp/src/app/ngx-restworld-client/models/client-settings.ts
import { ApiUrl } from "./api-url";

/**
 * These are the settings that the client is retrieving from its client-backend.
 * It is then used by the {@link SettingsService} to set everything up.
 */
export interface ClientSettings {
    /**
     * The URLs to the APIs that the client can call.
     */
    apiUrls: ApiUrl[];
    /**
     * Any extra settings that the client may need.
     * RESTworld itself does not need these, but get's them from the frontend-backend so you can transport additional when needed.
     */
    extensions: { [key: string]: string; }
}
